### PR TITLE
Removed the not ready enum type

### LINKED ISSUE #796: Remove `NotReady` variant for `PromiseError`
This variant cannot be constructed by the runtime, so no need to have it within our types. We can always add it back in the future if the functionality gets added.

[https://github.com/near/near-sdk-rs/blob/c0219b80de590945d8a203be046150c23f296cf9/near-sdk/src/types/vm_types.rs#L39](https://github.com/near/near-sdk-rs/blob/c0219b80de590945d8a203be046150c23f296cf9/near-sdk/src/types/vm_types.rs#L39)

### PR DESCRIPTION
**Pull request by DavidM-D**
It can't be generated by the runtime fixes #796

### CODE REVIEW

**Review by austinabell — APPROVED** [review]
I think this is the correct way to go as it's unclear what the semantics would even look like if this was to be handled from the application level.

Would love another opinion on this. Also, we probably want to hold this PR until we are ready to make a major version release as this is a breaking change

**Comment by itegulov** [comment]
Shouldn't we also remove it from https://github.com/near/near-sdk-rs/blob/master/near-sdk/src/types/vm_types.rs#L18?

**Comment by DavidM-D** [comment]
🤦‍♂️

**Pull request merged**
